js/util/util.js

```diff
@@ -474,3 +475,27 @@ exports.warnOnce = function(message) {
         warnOnceHistory[message] = true;
     }
 };
+
+/**
+ * Calculates centroid of all the points
+ * @param {Array} points
+ * @returns {Array} [x,y] Centroid of all the points
+ * @private
+ */
+exports.centroid = function(points) {
+    assert(points !== null);
+    assert(points.length > 0);
+
+    if (points.length === 1) {
+        return [points[0].x, points[0].y];
+    }
+    // This tries to prevent overflow/underflow of the float
+    var xC = points[0].x, yC = points[0].y;
+    var xSum = 0, ySum = 0, len = 0;
+    points.forEach(function(coord) {
+        xSum += coord.x - xC;
+        ySum += coord.y - yC;
+        len++;
+    });
```

Comment:
Also, I think we should use polygon centroid algorithm like this https://en.wikipedia.org/wiki/Centroid#Centroid_of_polygon, which will choose a better center.